services/backend/src/assignment-controller.js

```diff
@@ -28,75 +32,68 @@ async function getAssignment(req, res) {
 
     if (dbResponse.Item) {
         res.json(fromDbFormat(dbResponse.Item));
-    }
-    else {
+    } else {
         throw new Error('Assignment does not exist');
     }
 }
 
 async function getAssignments(req, res) {
-    const requestedBy = req.body.uid;
-
-    // im assuming query() works well to get all assignments
-    // i did not consider the case of getting a subset of assignments instead of all assignments
-    
+    const requestedBy = req.auth.uid;
 
     const dbResponse = await dynamo
-    .query({
-        TableName: 'assignments',
-        Key: { uid: requestedBy },
-    })
-    .promise();
-
-    // I tried to follow a similar format to getting a single assignment
-    // I want to add a check in case of no assingments existing
-    // again I am not sure if I need to use the helper function fromDbFormat
-    // not sure how to return all assignments from query() to frontend
-
-    if (dbResponse.Items) {
-        res.json(fromDbFormat(dbResponse.Item));
-    }
-    else {
-        throw new Error('No assignments exist');
-    }
-    
+        .scan({
+            TableName: 'assignments',
+            FilterExpression: 'uid = :uid',
+            ExpressionAttributeValues: {
+                ':uid': requestedBy,
+            },
+        })
+        .promise();
+
+    res.json(dbResponse.Items.map(fromDbFormat(dbResponse.Items)));
 }
 
-// I was not sure how to handle creation of assignments yet
-// would we be provided with the aid or would we need to generate an aid?
-// considering creation is triggered by route /assignments and not /assignments/:aid
+async function createAssignment(req, res) {
+    const aid = uuid();
 
-async function createAssignment(req, res) { }
+    const assignment = toDbFormat({ aid, ...req.body });
```

Comment:
```suggestion
    const assignment = toDbFormat(
        _.merge({}, req.body, { aid }, _.pick(req.auth, ['uid']))
    );
```
This will add `uid` from the request auth